src/os/bluestore/BlueStore.cc

```diff
@@ -8314,8 +8314,12 @@ void BlueStore::_dump_extent_map(ExtentMap &em, int log_level)
   }
 }
 
-void BlueStore::_dump_transaction(Transaction *t, int log_level)
+void BlueStore::_dump_transaction(Transaction* const t, const int log_level)
 {
+  if (! cct->_conf->subsys.should_gather(ceph_subsys_bluestore, log_level)) {
```

Comment:
If I understand correctly dout(log_level) is expanded to the following:
#define dout_impl(cct, sub, v)						\
  do {									\
  if (cct->_conf->subsys.should_gather(sub, v)) {			\
    if (0) {								\
      char __array[((v >= -1) && (v <= 200)) ? 0 : -1] __attribute__((unused)); \
    }									\
    static size_t _log_exp_length=80; \
    ceph::logging::Entry *_dout_e = cct->_log->create_entry(v, sub, &_log_exp_length);	\
    ostream _dout_os(&_dout_e->m_streambuf);				\
    CephContext *_dout_cct = cct;					\
    std::ostream* _dout = &_dout_os;

and hence there is no sense in prepending it with and additional check for should_gather.